examples/core/gated_gcn/main.py

```diff
@@ -0,0 +1,147 @@
+from dgl.nn.pytorch.glob import AvgPooling
+from ogb.graphproppred import collate_dgl, DglGraphPropPredDataset, Evaluator
+from ogb.graphproppred.mol_encoder import AtomEncoder, BondEncoder
+
+
+class GatedGCN(nn.Module):
+    def __init__(
+        self,
+        hid_dim,
+        out_dim,
+        num_layers,
+        dropout=0.2,
+        batch_norm=True,
+        residual=True,
+        activation=F.relu,
+    ):
+        super(GatedGCN, self).__init__()
+
+        self.num_layers = num_layers
+        self.dropout = dropout
+
+        self.node_encoder = AtomEncoder(hid_dim)
+        self.edge_encoder = BondEncoder(hid_dim)
+
+        self.layers = nn.ModuleList()
+        for _ in range(self.num_layers):
+            layer = GatedGCNConv(
+                input_feats=hid_dim,
+                edge_feats=hid_dim,
+                output_feats=hid_dim,
+                dropout=dropout,
+                batch_norm=batch_norm,
+                residual=residual,
+                activation=activation,
+            )
+            self.layers.append(layer)
+
+        self.pooling = AvgPooling()
+        self.output = nn.Linear(hid_dim, out_dim)
+
+    def forward(self, g, node_feat, edge_feat):
+        # Encode node and edge feature.
+        hv = self.node_encoder(node_feat)
+        he = self.edge_encoder(edge_feat)
+
+        # GatedGCNConv layers.
+        for layer in self.layers:
+            hv, he = layer(g, hv, he)
+
+        # Output project.
+        h_g = self.pooling(g, hv)
+
+        return self.output(h_g)
+
+
+def train(model, device, data_loader, opt, loss_fn):
+    model.train()
+    train_loss = []
+
+    for g, labels in data_loader:
+        g = g.to(device)
+        labels = labels.to(torch.float32).to(device)
+        logits = model(g, g.ndata["feat"], g.edata["feat"])
+        loss = loss_fn(logits, labels)
+        opt.zero_grad()
+        loss.backward()
+        opt.step()
+        train_loss.append(loss.item())
+
+    return sum(train_loss) / len(train_loss)
+
+
+@torch.no_grad()
+def evaluate(model, device, data_loader, evaluator):
+    model.eval()
+    y_true, y_pred = [], []
+
+    for g, labels in data_loader:
+        g = g.to(device)
+        logits = model(g, g.ndata["feat"], g.edata["feat"])
+        y_true.append(labels.detach().cpu())
+        y_pred.append(logits.detach().cpu())
+
+    y_true = torch.cat(y_true, dim=0).numpy()
+    y_pred = torch.cat(y_pred, dim=0).numpy()
+
+    return evaluator.eval({"y_true": y_true, "y_pred": y_pred})["rocauc"]
+
+
+def main():
+    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+    # Load ogb dataset & evaluator.
+    dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
+    evaluator = Evaluator(name="ogbg-molhiv")
+
+    n_classes = dataset.num_tasks
+
+    split_idx = dataset.get_idx_split()
+    train_loader = GraphDataLoader(
+        dataset[split_idx["train"]],
+        batch_size=32,
+        shuffle=True,
+        collate_fn=collate_dgl,
```

Comment:
Do you still need `collate_dgl`? Does the default collate function of `GraphDataLoader` work in this case?

Reply:
the default collate function works in this case, but I think we should remain collate_dgl here to be consist with other examples like https://github.com/dmlc/dgl/blob/master/examples/sparse/graph_transformer.py

Reply:
Then that example should probably be fixed too. The primary reason of using `GraphDataLoader` is to automatically handling batching for graph property prediction in most cases. `collate_dgl` was developed by the OGB team before `GraphDataLoader` is available. @rudongyu for awareness

Reply:
remove the collate_dgl